src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -1086,18 +1086,11 @@ private int FormatNumeric(long value, Span<byte> destination)
             {
                 return FormatOctal(value, destination);
             }
-            else if (value < 0)
-            {
-                // GNU format: store negative numbers in big endian format with leading '0xff' byte.
-                destination.Slice(0, Offset).Fill(0xff);
-                BinaryPrimitives.WriteInt64BigEndian(destination.Slice(Offset), value);
-                return Checksum(destination);
-            }
             else
             {
-                // GNU format: store positive numbers in big endian format with leading '0x80' byte.
-                destination.Slice(0, Offset).Fill(0);
-                destination[0] = 0x80;
+                // GNU format: store negative numbers in big endian format with leading '0xff' byte.
+                //             store positive numbers in big endian format with leading '0x80' byte.
+                BinaryPrimitives.WriteUInt32BigEndian(destination, value < 0 ? 0xffffffff : 1U << 31);
```

Comment:
I'm not sure I get why we also have this 32BigEndian call. Can you please describe it for my own education?

Reply:
We're filling a 12 byte field. The lower 8 bytes are the long. We need to use the appropriate value for the first 4 bytes depending on whether the value is negative or positive. I'm going to replace `1U << 31` with `0x80000000` as that better matches the comment above.